### ISSUE TITLE
RFC: Investigate auto-balancing queue

### ISSUE THREAD

**Issue by TimJentzsch** [GrafeasGroup/tor#323]
Once https://github.com/GrafeasGroup/blossom/issues/174 has been implemented, we can think about making the queue fetching process smarter.

In the long term, it might become more difficult to make the queue varied, providing a good balance of content while not flooding it with a single subreddit. Once we have the subreddits saved separately as sources, we have insight how many posts by each subreddit are in the queue.

The bot could automatically pull in less posts of a subreddit that already has many posts in the queue. If we give each source content tags we could also make sure that we have a good amount of text posts and memes, while still having other content types represented. It could even respond to the number of total posts in the queue, making sure that the queue isn't flooded when nobody is transcribing and that we always have enough posts (ofc that would defeat the purpose of CtQ).

Basically, instead of making the process static and manual with upvote filters, the bot could dynamically manage the queue.

Of course, this would not be trivial to implement. We'd need to make quite a lot of requests to Blossom, or potentially "count" the number of posts with `tor` itself for efficiency. We'd need to potentially categorize each subreddit with tags to make sure we have a good balance of content. We'd need to specify how we want to balance text posts with other contents. We'd need to cache when each sub has been queried last, to make sure that not all content in the queue is "stale". All of this might be hard to implement and hard to test. But it might be worth it in the long run, when we have a lot more volunteers and a lot of subs.
